OrbitQt/ConnectionConfiguration.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_CONNECTION_CONFIGURATION_H_
+#define ORBIT_QT_CONNECTION_CONFIGURATION_H_
+
+#include <memory>
+#include <utility>
+
+#include "DeploymentConfigurations.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitClientData/ProcessData.h"
+#include "OrbitClientServices/ProcessManager.h"
+#include "OrbitGgp/Instance.h"
+#include "OrbitSsh/Context.h"
+#include "OrbitSsh/Credentials.h"
+#include "grpcpp/channel.h"
+#include "servicedeploymanager.h"
+
+namespace orbit_qt {
+
+struct StadiaConnection {
```

Comment:
This shouldn't be a struct, but a class with private fields and proper getters. Structs should only be used for objects that carry data, and should not enforce any invariants on this data (https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes).

Reply:
After a conversation with Hennining, Its split up now into different classes. Some in this PR in Connections.h and some in #1526 (TargetConfiguration.h)